hera_cal/frf.py

```diff
@@ -10,22 +10,165 @@
+    max_frate_coeffs, 2-tuple float
+      Maximum fringe-rate coefficients for the model max_frate [mHz] = x1 * | EW_bl_len | [ m ] + x2."
+      These are used only if case == 'max_frate_coeffs'.
+    uvb: UVBeam object, optional
+     UVBeam object with model of the primary beam. Only used if case=='uvbeam'
+    percentile_low: float, optional
+        if uvb is provided, filter fringe rates that are larger then this percentile
+        in the histogram of beam squared powers for instantaneous fringe rates.
+        Only used if case == 'uvbeam' or case == 'frate_profiles'
+    percentile_high: float, optional
+        if uvb is provided, filter fringe rates that are smaller then this percentile
+        in the histogram of beam squared powers for instantaneous fringe rates.
+        only used if case == 'uvbeam' or case == 'frate_profiles'
+    fr_freq_skip: int, optional
+        bin fringe rates from every freq_skip channels.
+        default is 1 -> takes a long time. We recommend setting this to be larger.
+        only used if case == 'uvbeam'
+    dfr: float, optional.
+        spacing of fringe-rate grid used to perform binning and percentile calc
+        in units of mHz.
+        default is None -> set to 1 / (dtime * Ntime) of uvd.
+        only used of case == 'uvbeam' or case == 'frate_profiles'
+    nfr: float, optional.
+        number of points on fringe-rate grid to perform binning and percentile calc.
+        default is None -> set to uvd.Ntimes.
+        only used of case == 'uvbeam' or case == 'frate_profiles'
+    verbose: bool, optional, lots of outputs!
+    frate_profiles: dict object, optional
+        Dictionary mapping antpairpol keys to fringe-rate profiles.
+        centered on a grid of length nfr spaced by dfr centered at 0.
+        only used if case == 'frate_profiles'
+    filter_kwargs: see fourier_filter for a full list of filter_specific arguments.
+
+    Returns
+    -------
+    frate_centers: dict object,
+        Dictionary with the center fringe-rate of each baseline in to_filter in units of mHz.
+        keys are antpairpols.
+    frate_half_widths: dict object
+        Dictionary with the half widths of each fringe-rate window around the frate_centers in units of mHz.
+        keys are antpairpols
+    """
+    if keys is None:
+        keys = list(uvd.get_antpairpols())
+    if case == 'sky':
+        # if max_frate_coeffs is none and uvb is none, fringe-rate filter all modes that could be occupied by sky emission.
+        frate_centers, frate_half_widths = sky_frates(uvd, keys=keys, frate_standoff=frate_standoff,
+                                                      frate_width_multiplier=frate_width_multiplier, min_frate_half_width=min_frate_half_width)
+    elif case == 'max_frate_coeffs':
+        assert max_frate_coeffs is not None, "max_frate_coeffs must be provided if case='max_frate_coeffs'."
+        # if uvb is None and max_frate_coeffs is not None, use max_frate_coeffs.
```

Comment:
can get rid of this comment b/c case is no longer selected implicitly

Reply:
done.